File: megumi/src/main/java/com/erisu/cloud/megumi/remind/logic/RemindMeLogic.java
package com.erisu.cloud.megumi.remind.logic;

import com.baomidou.mybatisplus.core.conditions.query.QueryWrapper;
import com.erisu.cloud.megumi.remind.mapper.DevRemindMapper;
import com.erisu.cloud.megumi.remind.pojo.DevRemind;
import net.mamoe.mirai.message.data.Message;
import net.mamoe.mirai.message.data.MessageChain;
import net.mamoe.mirai.message.data.MessageChainBuilder;
import net.mamoe.mirai.message.data.PlainText;
import org.springframework.stereotype.Component;

import javax.annotation.Resource;
import java.util.List;

/**
 * @Description TODO
 * @Author alice
 * @Date 2021/6/15 9:01
 **/
@Component
public class RemindMeLogic {
    @Resource
    private DevRemindMapper remindMapper;

    public void addRemindMe(DevRemind remind) {
        remindMapper.insert(remind);
    }

    public Message getRemindMe(String groupId, String senderId) {
        QueryWrapper<DevRemind> remindQueryWrapper = new QueryWrapper<>();
        remindQueryWrapper.eq("group_id", groupId);
        remindQueryWrapper.eq("qq_id", senderId);
        List<DevRemind> devReminds = remindMapper.selectList(remindQueryWrapper);
        MessageChainBuilder builder = new MessageChainBuilder();
//        final int[] index = {1};
        devReminds.forEach(r -> {
            MessageChain messages = MessageChain.deserializeFromJsonString(r.getRemindJson());
            builder.append("☑️　messageId-")
                    .append(String.valueOf(r.getId())).append(":\n").append(messages).append("\n");
//            index[0]++;
        });
        return builder.build();
    }

    /**
     * @param remindMeId
     * @param groupId
     * @param userId
     * @return
     */
    public Message removeRemindMe(String remindMeId, String groupId, String userId) {
        QueryWrapper<DevRemind> remindQueryWrapper = new QueryWrapper<>();
        remindQueryWrapper.eq("group_id", groupId);
        remindQueryWrapper.eq("qq_id", userId);
        remindQueryWrapper.eq("id", remindMeId);
        if (remindMapper.delete(remindQueryWrapper) > 0) {
            return new PlainText("删除成功");
        } else {
            return new PlainText("删除失败");
        }
    }
}

File: megumi/src/main/java/com/erisu/cloud/megumi/remind/logic/RemindMeLogic.kt
package com.erisu.cloud.megumi.remind.logic

import com.baomidou.mybatisplus.core.conditions.query.QueryWrapper
import com.erisu.cloud.megumi.remind.mapper.DevRemindMapper
import com.erisu.cloud.megumi.remind.pojo.DevRemind
import com.erisu.cloud.megumi.util.StreamMessageUtil
import net.mamoe.mirai.contact.Group
import net.mamoe.mirai.message.data.*
import org.springframework.core.io.ClassPathResource
import org.springframework.stereotype.Component
import java.util.function.Consumer
import javax.annotation.Resource

/**
 *@Description remindMe相关逻辑
 *@Author alice
 *@Date 2021/6/23 16:48
 **/
@Component
class RemindMeLogic {
    @Resource
    private lateinit var remindMapper: DevRemindMapper

    fun addRemindMe(remind: DevRemind) {
        remindMapper.insert(remind)
    }

    fun getRemindMe(groupId: String?, senderId: String?): Message {
        val remindQueryWrapper = QueryWrapper<DevRemind>()
        remindQueryWrapper.eq("group_id", groupId).eq("qq_id", senderId)
        val devReminds = remindMapper.selectList(remindQueryWrapper)
        val messageChain = MessageChainBuilder()
        devReminds.forEach(Consumer { (id, _, _, remindJson) ->
            val messages = MessageChain.deserializeFromJsonString(remindJson)
            messageChain.append("✅　messageId-")
                .append(id.toString()).append(":\n").append(messages).append("\n")
        })
        return messageChain.build()
    }

    /**
     * @param remindMeId
     * @param groupId
     * @param userId
     * @return
     */
    suspend fun removeRemindMe(remindMeId: String?, group: Group, userId: String?): Message {
        val remindQueryWrapper = QueryWrapper<DevRemind>()
        remindQueryWrapper.eq("group_id", group.id).eq("qq_id", userId).eq("id", remindMeId)
        val resultList = remindMapper.selectList(remindQueryWrapper)
        val image = StreamMessageUtil.generateImage(group, ClassPathResource("emoticon/露娜笑.jpg").inputStream)
        return if (remindMapper.delete(remindQueryWrapper) > 0)
            messageChainOf(PlainText("完成了${resultList[0].remindJson}真了不起~"), image)
        else PlainText("删除失败")
    }

}